### PR TITLE
community: add ProjectGutenbergRetriever

### PR DESCRIPTION
**Pull request by ovuruska**
- **Description:** This PR adds a user-friendly abstraction for the `ProjectGutenbergRetriever` class. The PR covers the implementation, notebook and tests.

- **Issue:** N/A

- **Dependencies:** 
  - `requests`: Used for making HTTP requests to Project Gutenberg.

- **Twitter handle:** N/A

### CODE REVIEW

File: libs/community/langchain_community/retrievers/project_gutenberg.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+from typing import List
```
Comment (eyurtsev) [inline libs/community/langchain_community/retrievers/project_gutenberg.py:1]: Could you introduce this as a utility? Most of the utilities can be thought of as clients. Then users can reuse the client to either create a custom tool, create a retriever or create a document loader

**Review by eyurtsev — COMMENTED** [review]
Would you mind surfacing this as a client in utilities rather than as a retriever?

File: libs/community/langchain_community/retrievers/project_gutenberg.py (hunk 1 above)
Comment (ovuruska) [reply libs/community/langchain_community/retrievers/project_gutenberg.py:1]: Thanks for the feedback @eyurtsev

**Comment by eyurtsev** [comment]
@ovuruska I am going to close this PR, please free to re-open if you'd like to propose a client for project-gutenberg

**Comment by ovuruska** [comment]
Alright @eyurtsev . Thanks for noticing

**Pull request closed without merging**